kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java

```diff
@@ -778,20 +815,41 @@ protected String appendExtractionExpression(String type, String name, String id,
                         .append("')");
         } else if (type.equals("datetime-local")) {
             if (isShowTime) {
-                jsonTemplate.append("document.getElementById('")
+                jsonTemplate.append("getDate('")
                         .append(id)
-                        .append("')")
-                        .append(getExtractionValue(jsType));
+                        .append("',")
+                        .append(FUNCTION_MAPPING.get(formFieldType))
+                        .append(") ");
             } else {
-                jsonTemplate.append("getLocalDateWithoutTime('")
+                jsonTemplate.append("getDateWithoutTime('")
                         .append(id)
-                        .append("') ");
+                        .append("',")
+                        .append(FUNCTION_MAPPING.get(formFieldType))
+                        .append(") ");
             }
+        } else if (type.equals("datetime")) {
```

Comment:
here we could re-use the boolean method proposed above. It would merge ~4 else-if clauses into a single one.